sharktank/sharktank/ops/attention_impls.py

```diff
@@ -39,6 +39,20 @@ def _extract_linear_scale(t):
 
 
 def masked_flash_attention(q, k, v, a):
+    if (
```

Comment:
After chatting with Kyle about this, I think this is okay as a temporary thing because we're in a rush to support this, but it would be good to add a TODO suggesting we use the trampoline like every other op instead of calling this kernel directly

Reply:
I am not sure why we need a special handling for the case of all-replicated tensors. If this op were a proper registered op, you would get the all-replicated overload for free like it is for all other ops now.

Reply:
https://github.com/nod-ai/shark-ai/blob/c7c5f4d7973a521a2a48c2e9a518ce6cc1295aaa/sharktank/sharktank/ops/_registry.py#L350-L354

Reply:
Yes, the issue is that it's not being registered normally. It looks like it's being conditionally registered:
https://github.com/nod-ai/shark-ai/blob/14b4a97fc7871454495bb561b9ea45e421875989/sharktank/sharktank/ops/attention_impls.py#L102-L105

https://github.com/nod-ai/shark-ai/blob/14b4a97fc7871454495bb561b9ea45e421875989/sharktank/sharktank/examples/export_paged_llm_v1.py#L50-L53

I am not sure why we're conditionally overriding @dan-garvey

Reply:
I think the correct is to register `flash_attention` as a separate op. Also merge the implementation of `masked_flash_attention` into it. The presence of the mask argument should drive which is used.
Then the model config drives what is called at the call site. The argument `--attention-kernel` should be changing the model config. I suppose its purpose is to override the model config. We already have it.